cookbook/controller/upload_file.rst

```diff
@@ -164,15 +188,220 @@ There are some important things to consider in the code of the above controller:
+
+    .. code-block:: xml
+
+        <!-- app/config/config.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+            http://symfony.com/schema/dic/services/services-1.0.xsd"
+        >
+            <!-- ... -->
+
+            <service id="app.brochure_uploader" class="AppBundle\FileUploader">
+                <argument>%brochures_directory%</argument>
+            </service>
+        </container>
+
+    .. code-block:: php
+
+        // app/config/services.php
+        use Symfony\Component\DependencyInjection\Definition;
+
+        // ...
+        $container->setDefinition('app.brochure_uploader', new Definition(
+            'AppBundle\FileUploader',
+            array('%brochures_directory%')
+        ));
+
+Now you're ready to use this service in the controller::
+
+    // src/AppBundle/Controller/ProductController.php
+
+    // ...
+    public function newAction(Request $request)
+    {
+        // ...
+
+        if ($form->isValid()) {
+            $file = $product->getBrochure();
+            $fileName = $this->get('app.brochure_uploader')->upload($file);
+
+            $product->setBrochure($fileName);
+
+            // ...
+        }
+
+        // ...
+    }
+
+Using a Doctrine Listener
+-------------------------
+
+If you are using Doctrine to store the Product entity, you can create a
+:doc:`Doctrine listener </cookbook/doctrine/event_listeners_subscribers>` to
+automatically upload the file when persisting the entity::
+
+    // src/AppBundle/EventListener/BrochureUploadListener.php
+    namespace AppBundle\EventListener;
+
+    use Doctrine\ORM\Event\LifecycleEventArgs;
+    use AppBundle\Entity\Product;
+    use AppBundle\FileUploader;
+
+    class BrochureUploadListener
+    {
+        private $uploader;
+
+        public function __construct(FileUploader $uploader)
+        {
+            $this->uploader = $uploader;
+        }
+
+        public function prePersist(LifecycleEventArgs $args)
+        {
+            $entity = $args->getEntity();
+
+            $this->uploadFile($entity);
+        }
+
+        public function preUpdate(LifecycleEventArs $args)
+        {
+            $entity = $args->getEntity();
+
+            $this->uploadFile($entity);
+        }
+
+        private function uploadFile($entity)
+        {
+            // upload only works for Product entities
+            if (!$entity instanceof Product) {
+                return;
+            }
+
+            $file = $entity->getBrochure();
+
+            // only upload new files
+            if (!$file instanceof UploadedFile) {
+                return;
+            }
+
+            $fileName = $this->uploader->upload($file);
+            $entity->setBrochure($fileName);
+        }
+    }
+
+Now, register this class as a Doctrine listener:
```

Comment:
Hmm, will have to look into this. I think it might is (as doctrine doesn't discover the entity changed)